File: backend/src/main/java/com/anilyadav/ems/service/auth/EmployeeService.java
package com.anilyadav.ems.service.auth;

import com.anilyadav.ems.dto.request.EmployeeRequest;
import com.anilyadav.ems.dto.response.EmployeeResponse;

import java.util.List;

public interface EmployeeService {

    EmployeeResponse createEmployee(EmployeeRequest request);

    EmployeeResponse getEmployeeById(Long id);

    List<EmployeeResponse> getAllEmployees();

    EmployeeResponse updateEmployee(Long id, EmployeeRequest request);

    void deleteEmployee(Long id);

    EmployeeResponse getEmployeeByCode(String employeeCode);
}

File: frontend/src/services/employeeService.js
import {
  getAllEmployeesApi,
  getEmployeeByIdApi,
  createEmployeeApi,
  updateEmployeeApi,
  deleteEmployeeApi,
} from "../api/employeeApi";

export const getAllEmployees = async () => {
  const response = await getAllEmployeesApi();
  return response.data;
};

export const getEmployeeById = async (id) => {
  const response = await getEmployeeByIdApi(id);
  return response.data;
};

export const createEmployee = async (employee) => {
  const response = await createEmployeeApi(employee);
  return response.data;
};

export const updateEmployee = async (id, employee) => {
  const response = await updateEmployeeApi(id, employee);
  return response.data;
};

export const deleteEmployee = async (id) => {
  await deleteEmployeeApi(id);
};